Support HTTP transport

seagetch opened the pull request:
Added support for HTTP Transport.
I confirmed HTTP Transport can interact with Google gemini cli.

This patch was generated automatically by OpenAI codex.
If you don't like it, please ignore this patch.

This patch also added signal handling code.
This is needed to kill the MCP server process safely if it uses HTTP Transport.

seagetch committed:
Fix HTTP transport request handling

seagetch committed:
Merge pull request #1 from seagetch/codex/refactor-transport-interface-and-implement-httptransport

Improve HTTP transport request handling

seagetch committed:
Enable HTTP by default; fix HTTP transport for vibe.d 1.3.x; allow ping pre-initialize; handle notification responses correctly; update README instructions

seagetch committed:
feat(example, stdio): graceful shutdown on SIGINT/SIGTERM

- Add cross-platform signal handling to example app and stop server on Ctrl-C/TERM
- Make update thread daemon and stop on signal to not block exit
- Add MCPServer.stop() wrapper to close the transport cleanly
- stdio transport: break loop on closed/EOF/EINTR to exit promptly

seagetch committed:
fix(http): close listener and SSE clients cleanly on shutdown

- Notify and clear SSE clients during shutdown
- Stop HTTP listener from within event loop to ensure FD is released
- Also stop listening in run() scope(exit) to avoid eventcore active handle warnings

seagetch committed:
Comments in app.d is restored.

gtnoble commented:
Thank you for your contribution. However, it looks like your added tests are failing. Please investigate these and fix the errors.

```
Projects/MCP/d-mcp-server [2] ➜ git:(seagetch) ✗ dub test
             Generating test runner configuration 'mcp-test-library' for 'library' (library).
     Pre-gen Running commands for openssl
    Starting Performing "unittest" build using /usr/bin/dmd for x86_64.
  Up-to-date diet-ng 1.8.4: target for configuration [library] is up to date.
  Up-to-date during 0.3.0: target for configuration [default] is up to date.
  Up-to-date taggedalgebraic 1.0.1: target for configuration [library] is up to date.
  Up-to-date eventcore 0.9.37: target for configuration [epoll] is up to date.
  Up-to-date stdx-allocator 2.77.5: target for configuration [library] is up to date.
  Up-to-date vibe-container 1.7.0: target for configuration [library] is up to date.
  Up-to-date vibe-core 2.13.0: target for configuration [epoll] is up to date.
  Up-to-date vibe-inet:textfilter 1.2.0: target for configuration [library] is up to date.
  Up-to-date vibe-serialization 1.1.2: target for configuration [library] is up to date.
  Up-to-date vibe-stream 1.2.0: target for configuration [library] is up to date.
  Up-to-date vibe-inet 1.2.0: target for configuration [library] is up to date.
  Up-to-date vibe-inet:crypto 1.2.0: target for configuration [library] is up to date.
  Up-to-date vibe-stream:tls 1.2.0: target for configuration [openssl] is up to date.
  Up-to-date vibe-http 1.3.1: target for configuration [library] is up to date.
    Building mcp 1.0.1+commit.6.gad32a41: building configuration [mcp-test-library]
source/mcp/transport/http_test.d(24,13): Error: no property `run` for `http` of type `std.net.curl.HTTP`
/usr/include/dmd/phobos/std/net/curl.d(2385,1):        struct `HTTP` defined here
Error /usr/bin/dmd failed with exit code 1.
```

seagetch committed:
test(http): stabilize SSE unittest and ensure non-blocking shutdown

- Use ephemeral ports; wait for server readiness
- Wait for first SSE bytes; abort on ping and swallow curl write error
- Mark server thread as daemon and avoid joins to prevent hangs
- Minor: increase operationTimeout for SSE client

This prevents flaky timeouts and hanging tests while keeping SSE behavior intact.

seagetch commented:
Thank you for the review.
Now I checked the `dub test`, and http_test passed basic POST method test and SSE based message handling test.

gtnoble commented:
Merged into dev branch, but incomplete. The server doesn't exit when expected.

gtnoble commented:
Okay, I fixed the exiting issues. Thank you for your contribution!
